src/thunderbots/software/util/visualizer_messenger/visualizer_messenger.cpp

```diff
@@ -42,170 +86,115 @@ namespace Util
         if (elapsed_ms < DESIRED_PERIOD_MS)
             return;
 
-        // Check if publisher is initialized before publishing messages
-        if (!this->publisher)
+        // Lock the list of current websockets
+        ws_mutex.lock();
```

Comment:
when locking for the duration of a function (which we essentially are here), I think it's recommended to use an `std::lock_guard` instead. If you put the following at the top of the function, it will lock until the function ends:
```
    // Take ownership of list of websockets for the duration of this function
    std::lock_guard<std::mutex> websocket_list_lock(ws_mutex);
```

Reply:
Done